### ISSUE TITLE
Unable to expand tree after upgrade to Vaadin 7.7.16

### ISSUE THREAD

**Issue by mwajcht** [vaadin/tree-grid#34]
After upgrading Vaadin from 7.7.15 to 7.7.16 (currently the newest Vaadin 7 version), TreeGrid stopped to work correctly. Now the arrows are missing which makes it impossible to expand collapsed data.

The bug can be easily reproduced using treegrid-demo application. It works correctly when compiled using Vaadin version up to 7.7.15. When using Vaadin 7.7.16 it looks like in attachment.
![tree-grid-vaadin-7 7 16](https://user-images.githubusercontent.com/3501221/54113299-458f3e80-43e8-11e9-8291-99d7d6c550fe.png)

**Comment by daPhantom** [comment]
We're having some issues as well since Vaadin 7.7.16. We're styling the Tree Grid very heavily and the hierarchy column is not wrapped with a depth element anymore. It's basically just text:

How it used to be < Vaadin 7.7.16

```
<td class="v-grid-cell" style="height: 40px; width: 218px;" colspan="1">
    <div class="v-tree-grid-node depth-2" style="">
        <span class="v-tree-grid-expander"></span>
        <div class="gwt-HTML v-tree-grid-cell-content">Cell Content</div>
    </div>
</td>
```

How it is with >= Vaadin 7.7.16

```
<td class="v-grid-cell" style="height: 40px; width: 218px;" colspan="1">
    Cell Content
</td>
```

Any idea what's causing this? I saw some Grid related commits within Vaadins 7.7.16 release to increase performance.

https://github.com/vaadin/framework/commit/65419f2babba04bfb2cb879d34800153d1ced583
https://github.com/vaadin/framework/commit/eafd44672650e076fc4a43362e11b47ffb0dbff1

**Comment by TatuLund** [comment]
This has broken probably after this patch https://github.com/vaadin/framework/pull/11199/files#diff-4eb418dffbcdaba703ef2968ed43f16fR914 which delays some calculations and the TreeGrid uses JSNI to access this data https://github.com/vaadin/tree-grid/blob/master/treegrid-addon/src/main/java/org/vaadin/treegrid/client/TreeGridConnector.java#L82. The most likely solution is to delay accessing of the data here https://github.com/vaadin/tree-grid/blob/master/treegrid-addon/src/main/java/org/vaadin/treegrid/client/TreeGridConnector.java#L55. But that needs some trial and error.

**Comment by daPhantom** [comment]
Hey @TatuLund , thank you for looking into this issue and providing feedback.

Is there any chance you guys might release a compatible treegrid version for the current Vaadin version? 

We really want to utilize the new performance gains from the grid changes in 7.7.16! 

Best,
_Felix_

**Comment by TatuLund** [comment]
Yes, my theory was correct. I applied the fix and tested it in 7.7.17, and it works now. I made pull request.
